crates/common_utils/src/pii.rs

```diff
@@ -88,20 +98,103 @@ where
 }
 
 /// Email address
-#[derive(Debug)]
-pub struct Email;
+#[derive(serde::Serialize, serde::Deserialize, Debug, Clone, PartialEq, Eq, Default)]
+pub struct Email(
+    Secret<String>
+);
```

Comment:
How is the deserialization happening, if it's not using the `FromStr` implementation?